controller/controller.go

```diff
@@ -38,23 +44,28 @@ func DefaultConfig() Config {
 type Controller interface {
 	// Submit schedules a group on the configured fleet cluster. This is done by
 	// setting the state of the units in the group to loaded.
-	Submit(req Request) error
+	Submit(req Request) (*task.TaskObject, error)
```

Comment:
Do we want to define an `interface` in the `controller/` package, so the client doesn't need knowledge of the other packages?